src/project_model/workspace.rs

```diff
@@ -1,80 +1,143 @@
-use crate::bsp_types::mappings::build_target::new_bsp_build_target;
-use crate::bsp_types::BuildTarget;
-use crate::project_model::package_dependencies::{PackageDependency, PackageWithDependenciesIds};
-use cargo_metadata::{CargoOpt, Error, MetadataCommand, Package};
+use crate::bsp_types::mappings::build_target::build_target_id_from_name_and_path;
+use crate::bsp_types::{BuildTarget, BuildTargetIdentifier};
+use crate::project_model::package::BspPackage;
+use cargo_metadata::camino::Utf8PathBuf;
+use cargo_metadata::{CargoOpt, Error, MetadataCommand};
+use std::collections::HashMap;
+
+use log::error;
 use std::path::PathBuf;
+use std::rc::Rc;
+
+pub type TargetIdToPackageNameMap = HashMap<BuildTargetIdentifier, String>;
```

Comment:
Jesli podajemy w nazwie "To" to domyslnie chodzi o mape wiec nie trzeba na koncu dodawac "Map"